aiida_vasp/workflows/relax.py

```diff
@@ -0,0 +1,298 @@
+    if not (pos or volume or shape):
+        raise ValueError('relaxation workflow requires at least one of "positions", "cell shape", "cell volume" to be alowed to change.')
+    if not pos:
+        ibrion = Ibrion(ion_updates=IbrionFlags.NO_UPDATE)
+    else:
+        ibrion = Ibrion(ion_updates=IbrionFlags.IONIC_RELAXATION_CG)
+    isif = Isif(calculate_stress=IsifStressFlags.FULL, vary_positions=pos, vary_cell_shape=shape, vary_cell_volume=volume)
+    nsw = Nsw(value=200, ibrion=ibrion)
+    return ibrion, isif, nsw
+
+
+def encut_factor(volume):
+    if volume:
+        return 1.3
+    return 1
+
+
+def clean_incar_overrides(inputs):
+    """Make sure the incar overrides do not clash with parameters necessary for relaxation."""
+    overrides = AttributeDict(inputs.incar_add.get_dict())
+    if 'ibrion' in overrides:
+        raise ValueError('overriding IBRION not allowed, use relax.xxx inputs to control')
+    if 'isif' in overrides:
+        raise ValueError('overriding ISIF not allowed, use relax.xxx inputs to control')
+    if 'nsw' in overrides:
+        if inputs.relax.positions.value and overrides.nsw < 1:
+            raise ValueError('NSW (num ionic steps) was set to 0 but relaxing positions was requested')
+
+    return overrides
+
+
+def l2_norm(vector_a):
+    """L^2 norm for a line vector"""
+    return numpy.sqrt(numpy.dot(vector_a, vector_a.T))
+
+
+def compare_structures(structure_a, structure_b):
+    """Compare two StructreData objects A, B and return a delta (A - B) of the relevant properties."""
+    delta = AttributeDict()
+    volume_a = structure_a.get_cell_volume()
+    volume_b = structure_b.get_cell_volume()
+    delta.volume = volume_a - volume_b
+
+    pos_a = numpy.array([site.position for site in structure_a.sites])
+    pos_b = numpy.array([site.position for site in structure_b.sites])
+    delta.pos = pos_a - pos_b
+
+    site_vectors = [delta.pos[i, :] for i in range(delta.pos.shape[0])]
+    delta.pos_lengths = numpy.array([l2_norm(vector) for vector in site_vectors])
+
+    delta.cell = numpy.array(structure_a.cell) - numpy.array(structure_b.cell)
+    delta.cell_lengths = numpy.array(structure_a.cell_lengths) - numpy.array(structure_b.cell_lengths)
+    delta.cell_angles = numpy.array(structure_a.cell_angles) - numpy.array(structure_b.cell_angles)
+
+    return delta
+
+
+class VaspRelaxWf(WorkChain):
+    """Structure relaxation workchain for VASP."""
+
+    @classmethod
+    def define(cls, spec):
+        super(VaspRelaxWf, cls).define(spec)
+        spec.expose_inputs(VaspBaseWf, include=['code', 'structure', 'potcar_family', 'potcar_mapping', 'options'])
+        spec.input('kpoints.mesh', valid_type=get_data_class('array.kpoints'), required=False)
+        spec.input('kpoints.distance', valid_type=get_data_class('float'), required=False)
+        spec.input('incar_add', valid_type=get_data_class('parameter'), required=False)
```

Comment:
This is fine for now. However, I would say that we should make an `IncarComposer` and then we only carry that around and do `incarcomposer.modify('TAG', something)` at correction time if need be.